partest/src/main/scala/scala/tools/nsc/MainGenericRunner.scala

```diff
@@ -63,73 +61,39 @@ class MainGenericRunner {
     val logger = new ScalaConsoleLogger(Level.Warn)
     val jsConsole = new ScalaConsoleJSConsole
     val semantics = readSemantics()
-    val classpath = createClasspath(command)
+    val ir = (
+        loadIR(command.settings.classpathURLs) :+
+        runnerIR(command.thingToRun, command.arguments)
+    )
 
     val jsRunner = new MemVirtualJSFile("launcher.js")
       .withContent(s"PartestLauncher().launch();")
 
-    val env =
-      if (optMode == NoOpt) new RhinoJSEnv(semantics)
-      else new NodeJSEnv
-
-    val runClasspath = optMode match {
-      case NoOpt   => classpath
-      case FastOpt => fastOptimize(classpath, logger, semantics)
-      case FullOpt => fullOptimize(classpath, logger, semantics.optimized)
+    val linker = Linker.linker(semantics, OutputMode.Default,
+        withSourceMap = false, parallel = true, full = optMode == FullOpt)
+
+    val libJSEnv = {
+      if (optMode == NoOpt) {
+        val unit = linker.linkUnit(ir, logger)
+        val env = new RhinoJSEnv(withDOM = false).withSourceMap(false)
+        env.loadLibs(Nil, unit)
+      } else {
+        val output = WritableMemVirtualJSFile("partest-fastOpt.js")
+        linker.link(ir, output, logger)
+        new NodeJSEnv().loadLibs(Nil, output)
```

Comment:
I understand this mimics what happened before. But now it's pretty striking: it's weird to hard-code `NoOpt` equals `RhinoJSEnv` and `Fast/FullOpt` equals `NodeJSEnv`. We probably shouldn't change this behavior in this PR, but a comment would be welcome.